toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/panels/FileChoosePanel.java

```diff
@@ -0,0 +1,121 @@
+    private JButton fileSearch;
+
+    public FileChoosePanel(Properties properties, String strkey, String label, EnumError errorType) throws IOException {
+        super(properties,strkey);
+        this.label = label;
+        this.errorType = errorType;
+        super.setBasicProperties(strkey);
+    }
+
+    private void chooseDirectory(JTextField textField, String strKey) {
+        JFileChooser dialogue = new JFileChooser(textField.getText());
+        dialogue.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+        dialogue.setDialogTitle("Select directory for " + strKey);
+        dialogue.showOpenDialog(null);
+
+        if (dialogue.getSelectedFile() != null) {
+            if (dialogue.getSelectedFile().isDirectory()) {
+                LOG.info("File selected : " + dialogue.getSelectedFile().getAbsolutePath());
+                textField.setText(dialogue.getSelectedFile().getAbsolutePath());
+                try {
+                    testIconValid(false);
+                } catch (IOException e) {
+                    LOG.error(e.getMessage(), e);
+                }
+            }
+        }
+    }
+
+    private void chooseFile(JTextField textField) {
+        JFileChooser dialogue = new JFileChooser(textField.getText());
+        dialogue.setDialogTitle(UIMessages.SELECT_FILE);
+        dialogue.showOpenDialog(null);
+        dialogue.setMaximumSize(getMaximumSize());
+
+        if (dialogue.getSelectedFile() != null && dialogue.getSelectedFile().isFile()) {
+            LOG.info("File selected : " + dialogue.getSelectedFile().getAbsolutePath());
+            textField.setText(dialogue.getSelectedFile().getAbsolutePath());
+            try {
+                testIconValid(false);
+            } catch (IOException e) {
+                LOG.error(e.getMessage(), e);
+            }
+        }
+    }
+    private boolean testIconValidDirectory(boolean runTryValue) throws IOException {
+        boolean fileOrDirectory = false;
+        if(strkey.equals(AgentConfigProvider.TOAST_CHROMEDRIVER_PATH)) {
+            fileOrDirectory = true;
+        }
+        return ConfigTesterHelper.testWebAppDirectory(textField.getText(),runTryValue, fileOrDirectory);
+    }
+
+    public void testIconValid(boolean runTry) throws IOException {
+        boolean isNotKo = testIconValidDirectory(runTry);
+        String errorMess = UIMessages.ERROR_FILE_NOT_EXIST;
+        JLabel icon = iconNotValid;
+        if(isNotKo) {
+            LOG.info("Status of " + textField.getText() + " : OK");
+            icon = iconValid;
+            errorMess = " ";
+        } else {
+            LOG.info("Status of " + textField.getText() + " : KO");
+        }
+        iconPanel.removeAll();
+        iconPanel.add(icon);
+        errorLabel.setText(errorMess);
+        this.repaint();
+        this.revalidate();
+    }
+
+    public JButton createBasicFileSearch(JTextField textField, String strKey, Boolean fileOrDir) {
+        JButton fileSearch = new JButton();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "fileSearch" which hides the field declared at line 24. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)